qt/python/mantidqt/mantidqt/widgets/workspacedisplay/test/test_workspacedisplay_view.py

```diff
@@ -31,35 +36,60 @@ def create_test_workspace(workspace_name, ragged=False):
 class WorkspaceDisplayViewTest(unittest.TestCase, QtWidgetFinder):
     @classmethod
     def setUpClass(cls):
-        cls.non_ragged_workspace = create_test_workspace("non-ragged")
+        cls.workspace = create_test_workspace("non-ragged")
         cls.ragged_workspace = create_test_workspace("ragged", ragged=True)
 
     @classmethod
     def tearDownClass(cls):
         AnalysisDataService.clear()
 
+    def tearDown(self):
+        if not sip.isdeleted(self.presenter.container):
+            self.presenter.container.close()
+            QApplication.sendPostedEvents()
+
     def test_that_the_workspace_display_opens_and_closes_ok_with_a_non_ragged_workspace(self):
-        presenter = MatrixWorkspaceDisplay(self.non_ragged_workspace)
-        presenter.show_view()
-        self.assert_widget_created()
+        self.presenter = MatrixWorkspaceDisplay(self.workspace)
+        self.presenter.show_view()
 
-        presenter.container.close()
+        self.assert_widget_created()
 
+        self.presenter.container.close()
         QApplication.sendPostedEvents()
 
         self.assert_no_toplevel_widgets()
 
     def test_that_the_workspace_display_opens_and_closes_ok_with_a_ragged_workspace(self):
-        presenter = MatrixWorkspaceDisplay(self.ragged_workspace)
-        presenter.show_view()
-        self.assert_widget_created()
+        self.presenter = MatrixWorkspaceDisplay(self.ragged_workspace)
+        self.presenter.show_view()
 
-        presenter.container.close()
+        self.assert_widget_created()
 
+        self.presenter.container.close()
         QApplication.sendPostedEvents()
 
         self.assert_no_toplevel_widgets()
 
+    def test_that_copy_spectrum_to_table_does_not_cause_an_error_with_a_non_ragged_workspace(self):
+        with mock.patch('qtpy.QtCore.QItemSelectionModel.selectedRows') as patch:
+            self.presenter = MatrixWorkspaceDisplay(self.workspace)
+            self.presenter.show_view()
+            table = self.presenter.view.currentWidget()
+
+            patch.return_value = [table.model().index(0, 0)]
+
+            self.presenter.action_copy_spectrum_to_table(table)
```

Comment:
An addendum, in general are we happy to test for error not being tested implicitly rather than explicitly? See discussion here https://stackoverflow.com/questions/4319825/python-unittest-opposite-of-assertraises.

Reply:
It will be interesting to hear from a gatekeeper on this as I am not sure which is preferred